src/graph.rs

```diff
@@ -354,6 +302,78 @@ impl From<String> for Value {
     }
 }
 
+impl fmt::Display for Value {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Value::Null => write!(f, "#null"),
+            Value::Boolean(value) => write!(f, "{}", value),
+            Value::Integer(value) => write!(f, "{}", value),
+            Value::String(value) => write!(f, "\"{}\"", value),
+            Value::List(values) => {
+                write!(f, "[")?;
+                for elem in values {
+                    write!(f, "{},", elem)?;
+                }
+                write!(f, "]")
+            }
+            Value::Set(values) => {
+                write!(f, "{{")?;
+                for elem in values {
+                    write!(f, "{},", elem)?;
+                }
+                write!(f, "}}")
+            }
```

Comment:
Can we retain the behavior from before, where commas were used as separators instead of terminators?  (The language allows you to use them as terminators in DSL source, but that doesn't mean we have to use that in the pretty-printed output as well.)